### ISSUE TITLE
azuredevops_serviceendpoint_github with github app fails to plan

### ISSUE THREAD

**Issue by Yanson** [microsoft/terraform-provider-azuredevops#295]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and Azure DevOps Provider) Version

```
$ terraform -v
Terraform v0.14.5
```

```
terraform {
  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = "=2.43.0"
    }
    azuredevops = {
      source  = "microsoft/azuredevops"
      version = "0.1.1"
    }
  }
  required_version = ">= 0.14"
}
```

### Affected Resource(s)

* `azuredevops_serviceendpoint_github`

### Terraform Configuration Files

```hcl
resource "azuredevops_serviceendpoint_github" "serviceendpoint_gh_3" {
  project_id = azuredevops_project.project.id
  service_endpoint_name = "Sample GithHub Apps: Azure Pipelines"
  description = ""
}
```

### Expected Behavior

Plan should be successful.

### Actual Behavior

Validation fails:

```
Error: expected "description" to not be an empty string or whitespace
```

### Steps to Reproduce

1. `terraform import`
2. `terraform plan`

### Important Factoids

n/a

### References

[Docs](https://registry.terraform.io/providers/microsoft/azuredevops/latest/docs/resources/serviceendpoint_github) state `You must also set the description to "" explicitly.` however there is [validation](https://github.com/microsoft/terraform-provider-azuredevops/blob/4a93bd69a26d5380cb5761c329115e42e4e5cd6e/azuredevops/internal/service/serviceendpoint/commons.go#L71-L71) preventing that.

If setting a description or leaving it out (and picking up the default) it says:

```
Error: Error updating service endpoint in Azure DevOps: Github Apps can not be updated must match imported values exactly
```

**Comment by xuzhang3** [comment]
Hi @Yanson  This is a document error. In ADO `description` is an optional configuration and cannot be empty string.

**Comment by Yanson** [comment]
Yes, except "Github Apps do not support a description and will always be empty string". If the description is left off it defaults to "Managed by Terraform" and then the plan fails with "Github Apps can not be updated must match imported values exactly".

**Comment by xuzhang3** [comment]
Hi @Yanson  can you add the description for your service connections? The description was expect not to be an empty string in ADO provider.

**Comment by Yanson** [comment]
Hi @xuzhang3, I cannot set the description as terraform errors with:

> Error: Error updating service endpoint in Azure DevOps: Github Apps can not be updated must match imported values exactly

The description needs to be an empty string, but the provider prevents that.

**Comment by jeremywitte** [comment]
Importing an `azuredevops_serviceendpoint_github` resource backed by an existing GitHub App authentication worked as documented with the following provider version:
```hcl
terraform {
  required_providers {
    azuredevops = {
      source  = "terraform-providers/azuredevops"
      version = "0.0.1"
    }
  }
  required_version = ">= 0.13"
}
```
I suspect versions beginning with `0.1.0` included additional validation which broke the previous handling/detection of GitHub App auth based versions of this service connection resource.

Since the GitHub App-based auth version is distinctively different from `auth_personal` and `auth_oauth` versions, I would advocate for a more explicit declaration OR a *Data Source* to be utilized strictly for reference.

Potential explicit auth declaration:
```hcl
resource "azuredevops_serviceendpoint_github" "serviceendpoint_gh" {
  project_id = azuredevops_project.project.id
  service_endpoint_name = "Sample GithHub Apps: Azure Pipelines"
  description = ""  # Either require "" explicitly or do not specify at all

  auth_app = true
}
```

**Comment by xuzhang3** [comment]
Hi @jeremywitte @Yanson The validation was added after v0.0.1.  As portal accept empty string,  I will update the validation.

**Comment by xuzhang3** [comment]
Fixed in v0.1.3, closing this issue. Feel free to open another issue if you have questions.
